auth/test/test_basic.py

```diff
@@ -19,7 +19,8 @@
 def _token(username, password):
     assert isinstance(username, str)
     assert isinstance(password, str)
-    return "basic " + b64encode("%s:%s" % (username, password))
+    token_bytes = b"%s:%s" % (username.encode('ascii'), password.encode('ascii'))
```

Comment:
What happens if the username or password is in unicode? While we don't support unicode usernames, we should make sure we don't raise an exception. Can you add a test for that (if we don't have one)?